sources/models/src/lib.rs

```diff
@@ -530,9 +530,29 @@ struct OciDefaults {
 
 ///// The hard and soft limit values for an OCI defaults resource limit.
 #[model(add_option = false)]
+#[derive(Copy, Clone, Debug, Deserialize, Serialize, Eq, Ord, PartialOrd, PartialEq)]
 struct OciDefaultsResourceLimit {
-    hard_limit: u32,
-    soft_limit: u32,
+    #[serde(deserialize_with = "deserialize_limit")]
+    hard_limit: i64,
+    #[serde(deserialize_with = "deserialize_limit")]
+    soft_limit: i64,
+}
+
+impl OciDefaultsResourceLimit {
+    fn get_limit(limit: i64) -> u64 {
+        match limit {
+            -1 => u64::MAX,
+            _ => limit as u64,
+        }
+    }
```

Comment:
nit: I forget where I picked up this convention, maybe Java, but I prefer to put public functions first since they're the external interface for the type, and private functions later since they're implementation details.